waggle-dance-core/src/main/java/com/hotels/bdp/waggledance/mapping/service/impl/PrefixBasedDatabaseMappingService.java

```diff
@@ -181,7 +182,9 @@ private boolean includeInResults(MetaStoreMapping metaStoreMapping, String prefi
   @Override
   public DatabaseMapping databaseMapping(@NotNull String databaseName) {
     // Find a Metastore with a prefix
-    for (String metastorePrefix : mappingsByPrefix.keySet()) {
+    Iterator<Entry<String, DatabaseMapping>> iterator = mappingsByPrefix.entrySet().iterator();
+    while (iterator.hasNext()) {
+      String metastorePrefix = iterator.next().getKey();
```

Comment:
From the warning that you sent me for this change:

> makes inefficient use of keySet iterator instead of entrySet iterator
This method accesses the value of a Map entry, using a key that was retrieved from a keySet iterator. It is more efficient to use an iterator on the entrySet of the map, to avoid the Map.get(key) lookup.

So, to complete this optimization, you need to use entry.getValue() on line 189.

`Entry<String, DatabaseMapping> entry = iteratory.next();`
`DatabaseMapping databaseMapping = entry.getValue();`

Reply:
Yes, good spot.